From dfd38cb29c0768692f886d3ab9158bd2b3132582 Mon Sep 17 00:00:00 2001
From: Changqing Li <changqing.li@windriver.com>
Date: Tue, 22 Nov 2022 15:20:48 +0800
Subject: [PATCH] makefile: use conditional assignment for KBUILD_OUTPUT

Refer [1],from make 4.4, all variables that are marked as export will
also be passed to the shell started by the shell function. use "=" will
make KBUILD_OUTPUT always empty for shell function, use "?=" to make
"export KBUILD_OUTPUT" in enrironment can work.

[snip of 4.4 NEWS]
* WARNING: Backward-incompatibility!
   Previously makefile variables marked as export were not exported to commands
   started by the $(shell ...) function.  Now, all exported variables are
   exported to $(shell ...).
[snip]

[1] https://git.savannah.gnu.org/cgit/make.git/tree/NEWS?h=4.4&id=ed493f6c9116cc217b99c2cfa6a95f15803235a2#n74

Upstream-Status: Submitted [linuxptp-devel@lists.sourceforge.net]

Signed-off-by: Changqing Li <changqing.li@windriver.com>
---
 makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makefile b/makefile
index 529d8a0..3db60fa 100644
--- a/makefile
+++ b/makefile
@@ -15,7 +15,7 @@
 # with this program; if not, write to the Free Software Foundation, Inc.,
 # 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-KBUILD_OUTPUT =
+KBUILD_OUTPUT ?=
 
 DEBUG	=
 CC	?= $(CROSS_COMPILE)gcc
-- 
2.25.1

